app/questionnaire/rules/operations.py

```diff
@@ -0,0 +1,217 @@
+from copy import deepcopy
+from datetime import date
+from decimal import Decimal
+from typing import (
+    Iterable,
+    Mapping,
+    Optional,
+    Sequence,
+    Sized,
+    TypedDict,
+    TypeVar,
+    Union,
+)
+
+from babel.dates import format_datetime
+from dateutil.relativedelta import relativedelta
+
+from app.questionnaire.rules.helpers import ValueTypes, casefold
+from app.questionnaire.rules.operator import OPERATION_MAPPING
+from app.questionnaire.rules.utils import parse_datetime
+from app.questionnaire.value_source_resolver import ValueSourceTypes
+from app.settings import DEFAULT_LOCALE
+
+ComparableValue = TypeVar("ComparableValue", str, int, float, Decimal, date)
+NonArrayPrimitiveTypes = Union[str, int, float, Decimal, None]
+
+DAYS_OF_WEEK = {
+    "MONDAY": 0,
+    "TUESDAY": 1,
+    "WEDNESDAY": 2,
+    "THURSDAY": 3,
+    "FRIDAY": 4,
+    "SATURDAY": 5,
+    "SUNDAY": 6,
+}
+
+SELF_REFERENCE_KEY = "self"
+
+
+class DateOffset(TypedDict, total=False):
+    days: int
+    months: int
+    years: int
+    day_of_week: str
+
+
+class Operations:
+    """
+    A class to group the operations
+    """
+
+    def __init__(self, language: str):
+        self._language = language
+        self._locale = DEFAULT_LOCALE if language in ["en", "eo"] else language
```

Comment:
This is something we should make consistent, currently it is not. List is used when they don't need to be, mainly because it is negligible. I have changed since it is a better approach to use set. Have changed the one in placeholder transforms also.